### ISSUE TITLE
Make the CRD compliant with both k8s /v1beta1 and /v1

### ISSUE THREAD

**Issue by perfecthu** [projectcalico/calico#3445]
## Expected Behavior

The `apiextensions.k8s.io/v1beta1` version of `CustomResourceDefinition` is deprecated in Kubernetes `v1.16` and will no longer be served in `v1.19`. There should an initiative of calico CRD to move forward from `v1beta`  to `v1`

## Current Behavior

Here is one example CRD in calico in [`calico.yaml`](https://docs.projectcalico.org/manifests/calico.yaml)
```
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
   name: felixconfigurations.crd.projectcalico.org
spec:
  scope: Cluster
  group: crd.projectcalico.org
  version: v1
  names:
    kind: FelixConfiguration
    plural: felixconfigurations
    singular: felixconfiguration
```
All the CRDs in calico keep using `v1beta1` which is deprecated in k8s `v1.16`.

## Possible Solution

To make it compliant with older version k8s and work for new version, we could specify multiple versions in the CRD definition as follows. For more details, please refer Kubernetes docs [Specify multiple versions](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions)
```
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
   name: felixconfigurations.crd.projectcalico.org
spec:
  scope: Cluster
  group: crd.projectcalico.org
  # spec.version is deprecated in v1.16, use spec.versions instead
  versions:
    - name: v1beta1
      # Each version can be enabled/disabled by Served flag.
      served: true
      # One and only one version must be marked as the storage version.
      storage: true
      # a structural schema is mandatory for CustomResourceDefinitions in v1
      schema:
        openAPIV3Schema:
          type: object
          properties:
            {SPECIFIC_PROPERTIES}
    - name: v1
      served: true
      storage: false
      schema:
        openAPIV3Schema:
          type: object
          properties:
            {SPECIFIC_PROPERTIES}
  names:
    kind: FelixConfiguration
    plural: felixconfigurations
    singular: felixconfiguration
```

## Context

Please refer [Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/) for more details about how to support both older version and newer version of k8s.

## Your Environment

* Calico version: v3.13
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes v1.16

**Comment by tmjd** [comment]
Thank you for submitting this. We realize this is necessary and expect to begin working on this soon.

**Comment by caseydavenport** [comment]
This has been addressed in master and will be in Calico v3.15.0. 


PRs:
- https://github.com/projectcalico/libcalico-go/pull/1237
- https://github.com/projectcalico/calico/pull/3579
